docs/spec/generics.rst

```diff
@@ -1761,6 +1763,352 @@ overloads for each possible rank is, of course, a rather cumbersome
+
+   class Foo(Generic[DefaultStrT, T2]):
+       def __init__(self, a: DefaultStrT, b: T2) -> None: ...
+
+   reveal_type(Foo(1, ""))  # type is Foo[int, str]
+   Foo[int](1, "")          # Invalid: Foo[int, str] cannot be assigned to self: Foo[int, int] in Foo.__init__
+   Foo[int]("", 1)          # Invalid: Foo[str, int] cannot be assigned to self: Foo[int, int] in Foo.__init__
+
+When using a type parameter as the default to another type parameter, the
+following rules apply, where ``T1`` is the default for ``T2``.
+
+Scoping Rules
+~~~~~~~~~~~~~
+
+``T1`` must be used before ``T2`` in the parameter list of the generic.
+
+::
+
+   T2 = TypeVar("T2", default=T1)
+
+   class Foo(Generic[T1, T2]): ...   # Valid
+   class Foo(Generic[T1]):
+       class Bar(Generic[T2]): ...   # Valid
+
+   StartT = TypeVar("StartT", default="StopT")  # Swapped defaults around from previous example
+   StopT = TypeVar("StopT", default=int)
+   class slice(Generic[StartT, StopT, StepT]): ...
+                     # ^^^^^^ Invalid: ordering does not allow StopT to be bound
+
+Using a type parameter from an outer scope as a default is not supported.
+
+Bound Rules
+~~~~~~~~~~~
+
+``T1``'s bound must be a subtype of ``T2``'s bound.
+
+::
+
+   T1 = TypeVar("T1", bound=int)
+   TypeVar("Ok", default=T1, bound=float)     # Valid
+   TypeVar("AlsoOk", default=T1, bound=int)   # Valid
+   TypeVar("Invalid", default=T1, bound=str)  # Invalid: int is not a subtype of str
+
+Constraint Rules
+~~~~~~~~~~~~~~~~
+
+The constraints of ``T2`` must be a superset of the constraints of ``T1``.
+
+::
+
+   T1 = TypeVar("T1", bound=int)
+   TypeVar("Invalid", float, str, default=T1)         # Invalid: upper bound int is incompatible with constraints float or str
+
+   T1 = TypeVar("T1", int, str)
+   TypeVar("AlsoOk", int, str, bool, default=T1)      # Valid
+   TypeVar("AlsoInvalid", bool, complex, default=T1)  # Invalid: {bool, complex} is not a superset of {int, str}
+
+
+Type Parameters as Parameters to Generics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Type parameters are valid as parameters to generics inside of a
+``default`` when the first parameter is in scope as determined by the
+`previous section <scoping rules_>`_.
+
+::
+
+   T = TypeVar("T")
+   ListDefaultT = TypeVar("ListDefaultT", default=list[T])
+
+   class Bar(Generic[T, ListDefaultT]):
+       def __init__(self, x: T, y: ListDefaultT): ...
+
+   reveal_type(Bar)                    # type is type[Bar[T, ListDefaultT = list[T]]]
+   reveal_type(Bar[int])               # type is type[Bar[int, list[int]]]
+   reveal_type(Bar[int]())             # type is Bar[int, list[int]]
+   reveal_type(Bar[int, list[str]]())  # type is Bar[int, list[str]]
+   reveal_type(Bar[int, str]())        # type is Bar[int, str]
+
+Specialization Rules
+~~~~~~~~~~~~~~~~~~~~
+
+``Generic`` ``TypeAlias``\ es
+'''''''''''''''''''''''''''''
+
+``Generic`` ``TypeAlias``\ es can be further subscripted
```

Comment:
```suggestion
Generic type aliases
'''''''''''''''''''''''''

Generic type aliases can be further subscripted
```
"Generic" here refers to the concept of a generic, not the class `typing.Generic`, so it should not be monospaced. Similarly, this section should apply to all type aliases, whether or not they're created with `typing.TypeAlias`, so "TypeAlias" should also not be monospaced.